File: services/about-us.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { NgxSpinnerService } from 'ngx-spinner';
import { ToastrService } from 'ngx-toastr';

@Injectable({
  providedIn: 'root'
})
export class AboutUsService {

  constructor(private http:HttpClient,
    private spinner :NgxSpinnerService,private toastr:ToastrService, private router: Router ) { }
    data:any=[{}];
    display_Image:any;




  GetAllAboutUs(){
    //show spinner 
    this.spinner.show();
    //hits Api 
      this.http.get('https://localhost:44338/AboutUs/GetAllAboutUs/').subscribe((res)=>{
      this.data=res;
      this.spinner.hide();
      this.toastr.success('Data Retrieved !!');
    },err=>{
      this.spinner.hide();
      this.toastr.error(err.message, err.status);
    })

  }


  createAboutUs(body:any){

    this.spinner.show();
    body.image=this.display_Image;

debugger
    this.http.post('https://localhost:44338/AboutUs/CreateAboutUs/',body).subscribe((res)=>{
   debugger
      this.spinner.hide();
      this.toastr.success('saved Successfully :)');
    },error=>{
      this.spinner.hide();
      this.toastr.error(error.status,error.message);
    })
    window.location.reload();
  }







  uploadAttachment(file:FormData){
    
    this.http.post('https://localhost:44338/AboutUs/UploadImage/',file)
    .subscribe((res:any)=>{
      this.display_Image=res.image;
    },err=>{
      this.toastr.error(err.status,err.message);
    })
  }



  updateAboutUs(body:any){
    body.image=this.display_Image;
    
    this.http.put('https://localhost:44338/AboutUs/UpdateAboutUs/',body).subscribe((res)=>{
      this.toastr.success('updated Successfully :)');

    },err=>{
      this.toastr.error(err.status,err.message);
    })

  }


  delete(id:number){
    this.http.delete('https://localhost:44338/AboutUs/DeleteAboutUs/'+id).subscribe((res)=>{
      this.toastr.success('Deleted Successfully :)');
    },err=>{
      this.toastr.error(err.status,err.message);
    })
    window.location.reload();

  }






}

File: Service/AboutUsService.cs
using System;
using System.Collections.Generic;
using System.Text;
using Tahaluf.SoundCloud.Core.Data;
using Tahaluf.SoundCloud.Core.Repository;
using Tahaluf.SoundCloud.Core.Service;

namespace Tahaluf.SoundCloud.Infra.Service
{
    public class AboutUsService : IAboutUsService
    {

        private readonly IAboutUsRepository aboutUsRepository;

        public AboutUsService(IAboutUsRepository _aboutUsRepository)
        {
            aboutUsRepository = _aboutUsRepository;
        }



        public bool CreateAboutUs(AboutUs aboutUs)
        {
            return aboutUsRepository.CreateAboutUs(aboutUs);
        }

        public bool DeleteAboutUs(int id)
        {
            return aboutUsRepository.DeleteAboutUs(id);
        }

        public List<AboutUs> GetAllAboutUs()
        {
            return aboutUsRepository.GetAllAboutUs();
        }

        public bool UpdateAboutUs(AboutUs aboutUs)
        {
            return aboutUsRepository.UpdateAboutUs(aboutUs);
        }
    }
}
